// weav/build/elements/Bar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Bar = function Bar(props) {
  var percent = props.percent,
      style = props.style,
      barColor = props.barColor,
      textStyle = props.textStyle,
      text = props.text,
      textColor = props.textColor;


  var barWrapperStyle = {
    width: '100%',
    backgroundColor: barColor
  };
  var barStyle = {
    width: percent + '%'
  };
  var textDefaultText = {
    marginTop: 0,
    marginBottom: 0,
    color: textColor,
    fontFamily: 'sans-serif',
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 3,
    paddingBottom: 3
  };

  return _react2.default.createElement(
    'div',
    { style: barWrapperStyle },
    _react2.default.createElement(
      'div',
      { style: (0, _objectAssign2.default)({}, barStyle, style) },
      _react2.default.createElement(
        'p',
        { style: (0, _objectAssign2.default)({}, textDefaultText, textStyle, { color: textColor }) },
        text
      )
    )
  );
};

Bar.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  style: _propTypes2.default.object,
  percent: _propTypes2.default.number,
  barColor: _propTypes2.default.string,
  // eslint-disable-next-line react/forbid-prop-types
  textStyle: _propTypes2.default.object,
  text: _propTypes2.default.string,
  textColor: _propTypes2.default.string
};

Bar.defaultProps = {
  style: {},
  percent: 100,
  barColor: '#ffffff',
  textStyle: {},
  text: '',
  textColor: '#2E2E2E'
};

exports.default = Bar;
