// weav/build/elements/SocialButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var btnStyle = {
  paddingLeft: 15,
  paddingRight: 15,
  paddingTop: 10,
  paddingBottom: 10,
  textAlign: 'left',
  borderWidth: 0,
  borderRadius: 3,
  width: '100%'
};
var iconBasicStyle = {
  height: 20,
  display: 'inline-block',
  verticalAlign: 'middle'
};
var textBasicStyle = {
  display: 'inline-block',
  verticalAlign: 'middle',
  fontSize: 14,
  color: '#ffffff',
  paddingLeft: 8
};

var SocialButton = function SocialButton(props) {
  var style = props.style,
      text = props.text,
      icon = props.icon,
      iconStyle = props.iconStyle,
      backgroundColor = props.backgroundColor,
      onClick = props.onClick,
      altText = props.altText,
      textStyle = props.textStyle;


  return _react2.default.createElement(
    'button',
    {
      style: (0, _objectAssign2.default)(btnStyle, style, {
        backgroundColor: backgroundColor
      }),
      onClick: onClick
    },
    icon ? _react2.default.createElement('img', {
      src: icon,
      style: (0, _objectAssign2.default)(iconBasicStyle, iconStyle),
      alt: altText
    }) : null,
    _react2.default.createElement(
      'span',
      { style: (0, _objectAssign2.default)(textBasicStyle, textStyle) },
      text
    )
  );
};

SocialButton.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  style: _propTypes2.default.object,
  text: _propTypes2.default.string,
  altText: _propTypes2.default.string,
  onClick: _propTypes2.default.func,
  backgroundColor: _propTypes2.default.string,
  // eslint-disable-next-line react/forbid-prop-types
  icon: _propTypes2.default.any,
  // eslint-disable-next-line react/forbid-prop-types
  iconStyle: _propTypes2.default.object,
  // eslint-disable-next-line react/forbid-prop-types
  textStyle: _propTypes2.default.object
};

SocialButton.defaultProps = {
  style: {},
  text: '',
  altText: '',
  onClick: function onClick() {},
  backgroundColor: '',
  icon: null,
  iconStyle: {},
  textStyle: {}
};

exports.default = SocialButton;
