// weav/build/wrapper/panels/Panel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var panelStyle = {
  width: '100%',
  borderColor: '#BDBDBD',
  borderWidth: 1,
  borderStyle: 'solid',
  borderRadius: 5,
  paddingLeft: 15,
  paddingRight: 15,
  paddingTop: 10,
  paddingBottom: 10
};

var Panel = function Panel(props) {
  var children = props.children,
      style = props.style;


  return _react2.default.createElement(
    'div',
    { style: (0, _objectAssign2.default)(panelStyle, style) },
    children
  );
};

Panel.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  children: _propTypes2.default.any,
  // eslint-disable-next-line react/forbid-prop-types
  style: _propTypes2.default.object
};

Panel.defaultProps = {
  children: null,
  style: {}
};

exports.default = Panel;
