// weav/build/wrapper/panels/SocialSignInPanel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Panel = require('./Panel');

var _Panel2 = _interopRequireDefault(_Panel);

var _FacebookButton = require('../buttons/FacebookButton');

var _FacebookButton2 = _interopRequireDefault(_FacebookButton);

var _GithubButton = require('../buttons/GithubButton');

var _GithubButton2 = _interopRequireDefault(_GithubButton);

var _GoogleButton = require('../buttons/GoogleButton');

var _GoogleButton2 = _interopRequireDefault(_GoogleButton);

var _MediumButton = require('../buttons/MediumButton');

var _MediumButton2 = _interopRequireDefault(_MediumButton);

var _TwitterButton = require('../buttons/TwitterButton');

var _TwitterButton2 = _interopRequireDefault(_TwitterButton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var panelStyle = {
  paddingLeft: 15,
  paddingRight: 15,
  paddingTop: 10,
  paddingBottom: 10
};
var dividerStyle = {
  height: 10
};

var SocialSignInPanel = function SocialSignInPanel(props) {
  var onMediumClick = props.onMediumClick,
      onGithubClick = props.onGithubClick,
      onFacebookClick = props.onFacebookClick,
      onGoogleClick = props.onGoogleClick,
      onTwitterClick = props.onTwitterClick;


  return _react2.default.createElement(
    _Panel2.default,
    { style: panelStyle },
    _react2.default.createElement(_FacebookButton2.default, {
      onClick: onFacebookClick
    }),
    _react2.default.createElement('div', { style: dividerStyle }),
    _react2.default.createElement(_GithubButton2.default, {
      onClick: onGithubClick
    }),
    _react2.default.createElement('div', { style: dividerStyle }),
    _react2.default.createElement(_GoogleButton2.default, {
      onClick: onGoogleClick
    }),
    _react2.default.createElement('div', { style: dividerStyle }),
    _react2.default.createElement(_MediumButton2.default, {
      onClick: onMediumClick
    }),
    _react2.default.createElement('div', { style: dividerStyle }),
    _react2.default.createElement(_TwitterButton2.default, {
      onClick: onTwitterClick
    })
  );
};

SocialSignInPanel.propTypes = {
  onMediumClick: _propTypes2.default.func.isRequired,
  onGithubClick: _propTypes2.default.func.isRequired,
  onFacebookClick: _propTypes2.default.func.isRequired,
  onGoogleClick: _propTypes2.default.func.isRequired,
  onTwitterClick: _propTypes2.default.func.isRequired
};

exports.default = SocialSignInPanel;
