// weav/build/wrapper/buttons/Button.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

var _color = require('color');

var _color2 = _interopRequireDefault(_color);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Button = function Button(props) {
  var children = props.children,
      link = props.link,
      onClick = props.onClick,
      style = props.style,
      showBottomLine = props.showBottomLine,
      bottomLineSize = props.bottomLineSize,
      color = props.color,
      fontColor = props.fontColor,
      capitalize = props.capitalize;


  var buttonStyle = {
    color: fontColor,
    backgroundColor: color,
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 8,
    paddingBottom: 8,
    fontFamily: 'sans-serif',
    fontSize: 12,
    borderRadius: 5
  };

  var backgroundColorObj = (0, _color2.default)(color);
  var bottomLineStyle = {};
  if (showBottomLine) {
    bottomLineStyle = {
      borderColor: backgroundColorObj.lighten(0.3),
      borderBottomWidth: bottomLineSize,
      borderTopWidth: 0,
      borderLeftWidth: 0,
      borderRightWidth: 0,
      borderStyle: 'solid'
    };
  }

  if (capitalize) {
    buttonStyle.textTransform = 'uppercase';
  }

  if (link) {
    return _react2.default.createElement(
      'a',
      {
        href: link,
        style: (0, _objectAssign2.default)(buttonStyle, bottomLineStyle, style)
      },
      children
    );
  }

  return _react2.default.createElement(
    'button',
    {
      style: (0, _objectAssign2.default)(buttonStyle, bottomLineStyle, style),
      onClick: onClick
    },
    children
  );
};

Button.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  children: _propTypes2.default.any,
  // eslint-disable-next-line react/forbid-prop-types
  style: _propTypes2.default.object,
  link: _propTypes2.default.string,
  onClick: _propTypes2.default.func,
  color: _propTypes2.default.string,
  showBottomLine: _propTypes2.default.bool,
  bottomLineSize: _propTypes2.default.number,
  fontColor: _propTypes2.default.string,
  capitalize: _propTypes2.default.bool
};
Button.defaultProps = {
  children: null,
  style: {},
  link: null,
  onClick: function onClick() {},
  color: '#ffffff',
  showBottomLine: false,
  bottomLineSize: 2,
  fontColor: '#2E2E2E',
  capitalize: false
};

exports.default = Button;
