// weav/build/wrapper/buttons/GithubButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

// UI


var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

var _SocialButton = require('../../elements/SocialButton');

var _SocialButton2 = _interopRequireDefault(_SocialButton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// eslint-disable-next-line arrow-body-style
var GithubButton = function GithubButton(props) {
  var style = props.style;

  return _react2.default.createElement(_SocialButton2.default, _extends({}, props, {
    text: 'Sign in with Github',
    icon: 'https://raw.githubusercontent.com/keithweaver/weav/master/lib/assets/github_white_512.png',
    backgroundColor: '#444444',
    style: (0, _objectAssign2.default)({}, style, { borderWidth: 0 }),
    textStyle: {
      color: '#ffffff'
    }
  }));
};

exports.default = GithubButton;
